### PR TITLE
Libimagentrymarkdown/init

### PR DESCRIPTION
**Pull request by matthiasbeyer**
As #191 gets way to complex with all the markup languages implemented inside one crate, I opted for adding layers around single crates and implementing helper functionality upon them.

Later, #191 can be rewritten to use all the other crates to implement a unique interface for all of them.

---

This is my approach on adding some functionality in a wrapper around `hoedown` which might be useful for imag.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Would be nice to add some tests and I guess this is rather simple to test, actually.

File: libimagentrymarkdown/src/link.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+impl Render for LinkExtractor {
+
+    fn link(&mut self,
+            _: &mut Buffer,
+            _: Option<&Buffer>,
+            link: Option<&Buffer>,
+            title: Option<&Buffer>)
+        -> bool
+    {
+        let link  = link.and_then(|l| l.to_str().ok()).map(String::from);
```
Comment (TheNeikos) [inline libimagentrymarkdown/src/link.rs]: Please don't use a lowercase `l`, rather something like `x` or `c` or w/e

IlLi|

File: libimagentrymarkdown/src/link.rs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        let title = title.and_then(|l| l.to_str().ok()).map(String::from);
+
+        match (link, title) {
+            (Some(link), Some(title)) => {
+                self.links.push(Link { link: link, title: title });
+                true
```
Comment (TheNeikos) [inline libimagentrymarkdown/src/link.rs]: Going by the docs of hoedown you can return false here as you do not really render anything.

http://www.blaenkdenum.com/hoedown/src/hoedown/src/renderer/mod.rs.html#89-441

**Comment by matthiasbeyer** [comment]
I guess I will merge this now. Enhancements can be done in new branches then!

**Pull request merged**